packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -423,6 +432,10 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         }
     }
 
+    async executeCommand(commandOptions: CommandLineOptions): Promise<void> {
+        this.sendText(this.runInTerminalUtility.buildCommand(await this.processInfo, commandOptions) + '\n');
```

Comment:
Why is it async? We don't guarantee anything here for a client by awaiting. It should be `void`. `sendText` is `void` as well indicating that it does not return then text is processed.

Reply:
I don't know if `await this.processInfo` could fail, people could want to make sure that the call did not fail?

Reply:
ok, please add js-doc what resolve means, more important that it does not that command completed, for that they would need some other APIs